internals/osutil/user.go

```diff
@@ -124,3 +124,21 @@ func NormalizeUidGid(uid, gid *int, username, group string) (*int, *int, error)
 	}
 	return uid, gid, nil
 }
+
+// IsCurrent reports whether the given user ID and group ID are those of the
+// current user.
+func IsCurrent(uid, gid int) (bool, error) {
+	current, err := userCurrent()
+	if err != nil {
+		return false, err
+	}
+	currentUid, err := strconv.Atoi(current.Uid)
+	if err != nil {
```

Comment:
We might shortcut it with `err != nil || uid != currentUid`, in which case we might use the same pattern below and return a bare `true, nil` at the end.